### ISSUE TITLE
Close other pickers solutions for high amount of pickers

### ISSUE THREAD

**Issue by andrewgilliland** [hossein-zare/react-native-dropdown-picker#502]
Hi, I wanted to see if anyone had any other solutions to closing all other pickers when you have a high amount of pickers on one form (I have 32 on one form).

I have seen the solution here https://hossein-zare.github.io/react-native-dropdown-picker-website/docs/tutorials/close-other-pickers, which works great. But when you have a high amount of pickers, the amount of code grows quickly. Just wanted to get input or hear others ideas.

And thank you all contributors for making this library.

**Comment by mikehardy** [comment]
I haven't thought of anything fantastic yet. I tried to have an array of pickers that I dynamically added to and iterated over but - probably because I was mis-handling state and state / effect hooks - updating the state of a bunch of pickers in an array wasn't working for me. It really seemed like that would be the way though, and my pickers were dynamically generated so it should have worked. I'm curious if anyone has done anything like this either

**Comment by hossein-zare** [comment]
Hi @andrewgilliland 
You can create a custom component. (Not tested)

```jsx
// Not sure if React.memo could be used here
export default function CustomPicker(props) {
  const setOpen = useCallback((open) => {
    props.setOpenId(open ? props.id : null);
  }, []);

  return (
    <DropdownPicker
      {...props}
      open={props.openId === props.id}
      setOpen={setOpen}
    />
  );
}
```

Usage:
```jsx
const [openId, setOpenId] = setState(null);

<CustomPicker
  id={1}
  openId={openId}
  setOpenId={setOpenId}
  ...
/>

<CustomPicker
  id={2}
  openId={openId}
  setOpenId={setOpenId}
  ...
/>
```

**Comment by andrewgilliland** [comment]
@hossein-zare thank you! I did something slightly different, but this helped a lot. I made a similar callback function in the parent component (a form in this case) and one state object that has the value of open state of every dropdownpicker and associated it with a name prop (it could of just as well been an id) that was already being passed. Your example is cleaner though, and could do a refactor. I really appreciate the assistance!
